### DISCUSSION TITLE
Automated releases

### DISCUSSION THREAD

**Discussion by dbuezas** [dbuezas/lgt8fx#229] [General]
Apologies for disattending the pull requests for so long.
I am now introducing automatic releases each time a PR is merged into the repository. 

@LaZsolt, I registered as collaborator 🎉 . This should also allow you to merge Pull Requests.

Each time a PR is merged the release action will be triggered, which does the following
* Generate the release artefacts (the `lgt8f-1.0.x.zip` file)
* Update the `package_lgt8fx_index.json` file with the new version
* Commit and push those changes to the repository
* Generate a release

Except for PRs that only change the `readme.md` file, or files inside `/docs`. Those won't generate a release.

This means that there will be one release for each PR, so let's avoid merging a PR that is not fully functional.

**Comment by dbuezas** [comment]
My apologies if you are watching all activity of the repo, I have unwillingly spammed your email box with all the failed releases. Even though I created a separate repo to develop and test the releases, I still had to make some iterations in this repo to get it right (e.g in new repos, the "base" branch is called main instead of master).
I think everything is setup right now, so the unwanted emails should stop now.

**Reply by LaZsolt** [reply]
Thank you for invitating me to collaborate and for your trust.
I have some less significant issues and ideas in mind.

**Comment by dwillmore** [comment]
LOL, I don't mind!

**Comment by hmeijdam** [comment]
No problem and thanks to all for all the hard work. Much appreciated.
